core/templates/services/playthrough.service.ts

```diff
@@ -114,25 +116,35 @@ class CyclicStateTransitionsTracker {
 }
 
 class EarlyQuitTracker {
-  private stateName: string = null;
-  private expDurationInSecs: number = null;
+  private stateName: string = ' ';
+  private expDurationInSecs: number | null = null;
 
   foundAnIssue(): boolean {
     return (
       this.expDurationInSecs !== null &&
       this.expDurationInSecs < ServicesConstants.EARLY_QUIT_THRESHOLD_IN_SECS);
   }
 
-  recordExplorationQuit(stateName: string, expDurationInSecs: number): void {
-    this.stateName = stateName;
-    this.expDurationInSecs = expDurationInSecs;
+  recordExplorationQuit(
+      stateName: string,
+      expDurationInSecs: number | null
+  ): void {
+    if (this.stateName !== null && this.expDurationInSecs !== null) {
```

Comment:
Why might they be null?

Reply:
Here after changing type .I got the problem that i had to say that these values cant be null but i cant use non null operator so i added an if condition so that if ever it would be null it will throw an error.I disscussed it with @vojtechjelinek as well